### DISCUSSION TITLE
Adjust UI around developer profiles

### DISCUSSION THREAD

**Discussion by jacobdaddario** [joemasilotti/railsdevs.com#197] [Ideas]
I finally got around to making a profile on the site now that the developer profile can be marked as "Not looking for work". Originally I was hesitant because I didn't want to give the wrong impression to my employer, so thanks for adding this feature!

While making the developer profile I noticed a few things:
1. The copy of `My developer profile` in the menu under the profile picture feels kind of misleading. It feels like it should take me to my own profile, but instead it takes me to the form to edit my profile. I think we could change this to link to the profile and then present an option to edit the profile to the owning user on the profile itself. I think that this would be consistent with the UI on other websites.
2. The feedback on the direct upload of the avatar and cover image aren't visible with how long the form is now. I think it might be worth adding a UI element that maybe appears and sticks to the bottom of the screen once direct uploads have been initiated.
3. Some feedback on the submit button is missing. I think the cursor needs to be changed when the button is disabled, and maybe we could add in a UI element to the button to indicate progress. [This Tailwindcss example is what comes to mind for me.](https://tailwindcss.com/docs/animation#spin)

I'd be happy to pluck along on these ideas if you think they're a good idea, @joemasilotti. They're non-blocking to the features of the site, so I think they might be a good thing for me to contribute with without disrupting what you're doing.

**Comment by joemasilotti** [comment]
Thanks for taking the lead on this, and welcome to the site 😜

1. Great call, I agree. We can use the space normally reserved for the `[briefcase] Hire me` button and replace it with `[pencil] Edit` if the user owns the visible profile.
2. Let's punt on this for now. I think that 3 will help get the message across that something is going on. Then we can revisit.
3. Ohh, I like that! Some thoughts to follow.

---

Regarding the UX when submitting the form, there are a few things I'd like to call out.

1. Some form's submit buttons still need the cursor set (like #180).
2. Back before Turbo we could handle this via `data-disable-with` to set button copy and custom classes. I'm not 100% sure where we are with Turbo and this right now, but I'd love to handle all of this without any custom JavaScript, if possible.
3. That Tailwind example looks perfect to me, with one exception. When disabled, hovering shouldn't change the background color.
4. There's definitely an extraction here, but I'm not sure if its a component or something lighter weight like `@apply` on a single class. I'll leave that up to you!
